module/VuFind/src/VuFind/Controller/ContentController.php

```diff
@@ -55,28 +55,45 @@ public function contentAction()
             ->getLocale();
         $defaultLanguage = $this->getConfig()->Site->language;
 
+        $types = [
+            'phtml',
+            'md',
+        ];
+
         // Try to find a template using
         // 1.) Current language suffix
         // 2.) Default language suffix
         // 3.) No language suffix
-        $currentTpl = "templates/content/{$page}_$language.phtml";
-        $defaultTpl = "templates/content/{$page}_$defaultLanguage.phtml";
-        if (null !== $themeInfo->findContainingTheme($currentTpl)) {
-            $page = "{$page}_$language";
-        } elseif (null !== $themeInfo->findContainingTheme($defaultTpl)) {
-            $page = "{$page}_$defaultLanguage";
+        $templates = [
+            "{$page}_$language",
+            "{$page}_$defaultLanguage",
+            $page,
+        ];
+
+        $pathPrefix = "templates/content/";
+
+        foreach ($templates as $template) {
+            foreach ($types as $type) {
+                $filename = "$pathPrefix$template.$type";
+                $path = $themeInfo->findContainingTheme($filename, true);
+                if (null != $path) {
+                    $page = $template;
+                    $renderer = $type;
+                    break 2;
+                }
+            }
         }
 
-        if (empty($page) || 'content' === $page
-            || null === $themeInfo->findContainingTheme(
-                "templates/content/$page.phtml"
-            )
-        ) {
-            return $this->notFoundAction($this->getResponse());
+        if ($renderer === 'phtml') {
```

Comment:
To add to flexibility, as a complement to the suggested "class property listing legal types" idea, what if you create a separate method to build the view for each type, and then load it via convention -- e.g.:

<pre>
$method = 'getViewFor' . ucwords($renderer);
if (is_callable([$this, $method])) {
    return $this->$method($page, $path);
}
</pre>

(Not sure if there's a better way to pass parameters more consistently across different methods...)

Reply:
This does make sense, I'll do it that way.